File: 二叉树的层次遍历/C++/original.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 * 思路和python的广度优先遍历一样。
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> res;
        if(root == NULL) return res;
        vector<TreeNode*> q;
        q.push_back(root);
        while(q.size()!=0){
            vector<int> p;
            for(int i=0; i<q.size(); ++i){
                p.push_back(q[i]->val);
            }
            res.push_back(p);
            vector<TreeNode*> new_q;
            for(int j=0; j<q.size(); ++j){
                if(q[j]->left) new_q.push_back(q[j]->left);
                if(q[j]->right) new_q.push_back(q[j]->right);
            }
            q=new_q;
        }
        return res;
    }
};

File: 中级算法/树和图/二叉搜索树中第K小的元素/python/original.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    '''
    思路：通过递归把二叉树中的所有值保存在一个列表中，然后把列表排序，
    返回列表对应值即可。
    '''
    def kthSmallest(self, root: TreeNode, k: int) -> int:
        result = self.recursion(root)
        result.sort()
        return result[k-1]
    def recursion(self,root):
        if root == None:
            return []
        return [root.val]+self.recursion(root.right)+self.recursion(root.left)